### PR TITLE
Testing NSM with IPv6 payloads

### PR DESCRIPTION
**Pull request by rdimitrov**
Related to #920
## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/integration/basic_memif_test.go [hunk 1]
```diff
@@ -26,8 +27,9 @@ func TestSimpleMemifConnection(t *testing.T) {
 	k8s.PrepareDefault()
 
 	nodes := nsmd_test_utils.SetupNodesConfig(k8s, 1, defaultTimeout, []*pods.NSMgrPodConfig{})
+	useIPv4 := true
```
Comment (ivanayov) [inline test/integration/basic_memif_test.go]: Can you make it configurable? 
This way the tests could run for both IPv4 and 6, when we decide to test IPv6

File: controlplane/pkg/prefix_pool/prefixpool_test.go [hunk 2]
```diff
@@ -64,6 +64,23 @@ func TestNetExtract1(t *testing.T) {
 	Expect(err).To(BeNil())
 }
 
+func TestNetExtract2(t *testing.T) {
```
Comment (nickolaev) [inline controlplane/pkg/prefix_pool/prefixpool_test.go]: Please rename these to `TestNetExtractIPv4` and `TestNetExtractIPv6`.

File: sdk/endpoint/ipam.go [hunk 3]
```diff
@@ -57,8 +58,13 @@ func (ice *IpamEndpoint) Request(ctx context.Context, request *networkservice.Ne
 		return nil, err
 	}
 
-	//TODO: We need to somehow support IPv6.
-	srcIP, dstIP, prefixes, err := ice.prefixPool.Extract(request.Connection.Id, connectioncontext.IpFamily_IPV4, request.Connection.Context.ExtraPrefixRequest...)
+	/* Quick way to determine whether the pool is IPv4 or IPv6 */
+	currentIPFamily := connectioncontext.IpFamily_IPV4
+	if strings.Contains(ice.prefixPool.GetPrefixes()[0], ":") {
```
Comment (nickolaev) [inline sdk/endpoint/ipam.go]: That shoudl be OK, but to be even more sure, you should count the `:` and have at least 2 of them for a valid IPv6.

File: test/integration/basic_excluded_prefixes_test.go [hunk 4]
```diff
@@ -22,6 +22,8 @@ func TestExcludePrefixCheck(t *testing.T) {
 		return
 	}
 
+	nsmd_test_utils.Init()
```
Comment (nickolaev) [inline test/integration/basic_excluded_prefixes_test.go]: Can e mbed this in `NewK8s`?

File: sdk/endpoint/ipam.go [hunk 5]
```diff
@@ -57,8 +58,13 @@ func (ice *IpamEndpoint) Request(ctx context.Context, request *networkservice.Ne
...
+		currentIPFamily = connectioncontext.IpFamily_IPV6
+	}
+
+	srcIP, dstIP, prefixes, err := ice.prefixPool.Extract(request.Connection.Id, currentIPFamily, request.Connection.Context.ExtraPrefixRequest...)
```
Comment (nickolaev) [inline sdk/endpoint/ipam.go]: You shoudl also consider the IP family furhter down where the route to DNS `8.8.8.8` is set.

File: controlplane/pkg/prefix_pool/prefixpool_test.go (hunk 2 above)
Comment (rdimitrov) [reply controlplane/pkg/prefix_pool/prefixpool_test.go]: Resolved 👍

File: sdk/endpoint/ipam.go (hunk 3 above)
Comment (rdimitrov) [reply sdk/endpoint/ipam.go]: Resolved 👍

File: test/integration/basic_excluded_prefixes_test.go (hunk 4 above)
Comment (rdimitrov) [reply test/integration/basic_excluded_prefixes_test.go]: Resolved 👍

File: sdk/endpoint/ipam.go (hunk 5 above)
Comment (rdimitrov) [reply sdk/endpoint/ipam.go]: Resolved 👍

**Comment by rdimitrov** [comment]
Probably blocked due to disabled IPv6 support in both NSC/NSE containers. For more details, refer to the issue - https://github.com/networkservicemesh/networkservicemesh/issues/920#issuecomment-489155718

**Pull request merged**